compiler/ifa/num.h

```diff
@@ -79,6 +79,10 @@ class Immediate { public:
   uint32_t const_kind;
   uint32_t num_index;
   union {
+    // Unions are initalized based off the first element, so we need to have
+    // the largest thing first to make sure it is all zero initalized
+    complex128 v_complex128;
+    complex64  v_complex64;
```

Comment:
Yeah.  My ideal organization would put ints and uints after that, in decreasing size order.  I think that leaves booleans and strings at the end